internal/cli/opsmanager/clusters/delete.go

```diff
@@ -43,27 +45,149 @@ func (opts *DeleteOpts) Run() error {
 	if !opts.Confirm {
 		return nil
 	}
+
+	hostIds, err := opts.hostIDs()
+	if err != nil {
+		return err
+	}
+
+	// shutdown cluster
+	err = opts.shutdownCluster()
+	if err != nil {
+		return err
+	}
+
+	// Remove cluster from automation
+	err = opts.removeClusterFromAutomation()
+	if err != nil {
+		return err
+	}
+
+	// Stop monitoring
+	err = opts.stopMonitoring(hostIds)
+	if err != nil {
+		return err
+	}
+
+	fmt.Print("Cluster deleted\n")
+	return nil
+}
+
+func (opts *DeleteOpts) hostIDs() ([]string, error) {
 	current, err := opts.store.GetAutomationConfig(opts.ConfigProjectID())
+	if err != nil {
+		return nil, err
+	}
+
+	hostnameMap := make(map[string][]int32)
+	opts.replicaSetHostNames(current, hostnameMap, opts.Entry)
+	opts.shardClusterHostNames(current, hostnameMap)
+
+	var hostIds []string
+	for k, ports := range hostnameMap {
+		for _, port := range ports {
+			host, err := opts.store.HostByHostname(opts.ConfigProjectID(), k, int(port))
+			if err != nil {
+				return nil, err
+			}
+			hostIds = append(hostIds, host.ID)
+		}
+	}
 
+	if len(hostIds) == 0 {
+		return nil, fmt.Errorf("cluster '%s' doesn't exist", opts.Entry)
+	}
+
+	return hostIds, nil
+}
+
+func (opts *DeleteOpts) replicaSetHostNames(automationConfig *opsmngr.AutomationConfig, hostnameMap map[string][]int32, name string) {
```

Comment:
[nit] given you don't need `opts`
```suggestion
func replicaSetHostNames(automationConfig *opsmngr.AutomationConfig, hostnameMap map[string][]int32, name string) {
```